syntax = "proto3";
package remote_tcl;

service RemoteTcl {
    // 我们rpc服务的名字
    // 后面
    // 服务端 会用到 <ComputeServicer>
    // 客户端 会用到 <ComputeStub>

    rpc tcl (TclRequest) returns (TclResponse) {}
    rpc get_file (GetFileRequest) returns (GetFileResponse) {}
    rpc put_file (PutFileRequest) returns (PutFileResponse) {}
}

message Common {
    int32               stat = 1;
    optional int32      err = 2;
    optional string     err_info = 3;
}

message TclRequest {
    Common  common = 1;
    int32   timeout = 2;
    bool    raw = 3;
    string  cmd = 4;
}

message TclResponse {
    Common  common = 1;
    int32   timeout = 2;
    bool    raw = 3;
    string  cmd = 4;
    string  output = 5;
}

message PutFileRequest {
    Common  common = 1;
    string  src_path = 3;
    string  dst_path = 4;
    int32   size = 5;
    bytes   content = 6;
}

message PutFileResponse {
    Common  common = 1;
    string  src_path = 3;
    string  dst_path = 4;
    int32   size = 5;
}

message GetFileRequest {
    Common  common = 1;
    string  src_path = 2;
    string  dst_path = 3;
}

message GetFileResponse {
    Common  common = 1;
    string  src_path = 3;
    string  dst_path = 4;
    int32   size = 5;
    bytes   content = 6;
}